evm/src/hosts/EvmHost.sol

```diff
@@ -548,6 +579,12 @@ abstract contract EvmHost is IIsmpHost, IHostManager, Context {
      */
     function setFrozenState(bool newState) public onlyAdmin {
         _frozen = newState;
+
```

Comment:
Small optimization recommendation.

Consider emitting one single event: `HostFrozen(bool _state)` with the boolean parameter. There is not much need for the conditional logic for event emission considering the binary state of `_frozen`.